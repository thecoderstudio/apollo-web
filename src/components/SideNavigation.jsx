import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import media from '../util/media';

const propTypes = {
  children: PropTypes.node.isRequired
};

const Navigation = styled.div`
  max-width: 225px;
  width: 100%;
  background-color: ${props => props.theme.lightBlack};
  box-shadow: inset ${props => props.theme.boxShadow};
  padding-top: 32px;
  transition: 0.75s ease-in-out;
  overflow: hidden;

  ${
    media.phone`
      width: 45px;
    `
  }
`;

export default function SideNavigation(props) {
  const { className, children } = props;

  return (
    <Navigation className={className}>
      {children}
    </Navigation>
  );
}

SideNavigation.propTypes = propTypes;
