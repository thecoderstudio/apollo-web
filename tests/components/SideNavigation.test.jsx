import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SideNavigation from '../../src/components/SideNavigation';

function getComponent(props) {
  return renderer.create(
    <BrowserRouter>
      <SideNavigation {...props} />
    </BrowserRouter>
  );
}

describe('Settings side navigation', () => {
  let props;

  beforeEach(() => {
    props = {
      location: { pathname: '/path' }
    };
  });

  it('renders correctly', () => {
    let tree = getComponent(props).toJSON();
    expect(tree).toMatchSnapshot();

    tree = getComponent({
      location: { pathname: '/path' }
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
