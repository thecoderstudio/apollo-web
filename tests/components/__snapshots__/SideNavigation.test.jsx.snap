// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings side navigation renders correctly 1`] = `
.c0 {
  max-width: 225px;
  width: 100%;
  box-shadow: inset;
  padding-top: 32px;
  -webkit-transition: 0.75s ease-in-out;
  transition: 0.75s ease-in-out;
  overflow: hidden;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  pointer-events: inherit;
}

.c1:hover {
  opacity: 0.9;
}

.c3 {
  cursor: pointer;
}

.c4 {
  margin: 0 8px;
}

.c2 {
  padding: 12px;
  margin-top: 24px;
  border-radius: 4px;
  width: 100%;
  float: left;
  overflow: hidden;
  text-overflow: ellipsis;
  box-sizing: border-box;
  white-space: nowrap;
}

.c5 {
  margin: 0 8px;
}

@media (max-width:47.9375em) {
  .c0 {
    width: 45px;
  }
}

@media (max-width:47.9375em) {
  .c4 {
    margin: 0 auto;
  }
}

@media (max-width:47.9375em) {
  .c5 {
    display: none;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    disabled={false}
    href="/settings/test"
    onClick={[Function]}
  >
    <i
      className="c3 c4 fas fa-user-cog"
    />
    <span
      className="c5"
    >
      Settings
    </span>
  </a>
</div>
`;

exports[`Settings side navigation renders correctly 2`] = `
.c0 {
  max-width: 225px;
  width: 100%;
  box-shadow: inset;
  padding-top: 32px;
  -webkit-transition: 0.75s ease-in-out;
  transition: 0.75s ease-in-out;
  overflow: hidden;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  pointer-events: inherit;
}

.c1:hover {
  opacity: 0.9;
}

.c3 {
  cursor: pointer;
}

.c4 {
  margin: 0 8px;
}

.c2 {
  padding: 12px;
  margin-top: 24px;
  border-radius: 4px;
  width: 100%;
  float: left;
  overflow: hidden;
  text-overflow: ellipsis;
  box-sizing: border-box;
  white-space: nowrap;
}

.c5 {
  margin: 0 8px;
}

@media (max-width:47.9375em) {
  .c0 {
    width: 45px;
  }
}

@media (max-width:47.9375em) {
  .c4 {
    margin: 0 auto;
  }
}

@media (max-width:47.9375em) {
  .c5 {
    display: none;
  }
}

<div
  className="c0"
>
  <a
    className="c1 c2"
    disabled={false}
    href="/settings/test"
    onClick={[Function]}
  >
    <i
      className="c3 c4 fas fa-user-cog"
    />
    <span
      className="c5"
    >
      Settings
    </span>
  </a>
</div>
`;
